### PR TITLE
treewide: add targetPrefix to hardcoded references to CC=cc

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
While working on fixing cross-compilation of some packages, I noticed that over and over I had to replace `CC=cc` with `CC=${stdenv.cc.targetPrefix}cc` so that it would use the cross-compiler. I have made this change treewide.

Should not affect `build -> build` compilation since the prefix becomes the empty string.

Replicate with

```ShellSession
$ find pkgs -type f -name '*nix' | xargs sed -i '' 's/CXX=c\+\+/CXX=\$\{stdenv\.cc\.targetPrefix\}c\+\+/g'
$ find pkgs -type f -name '*nix' | xargs sed -i '' 's/LD=cc/LD=\$\{stdenv\.cc\.targetPrefix\}cc/g'
$ find pkgs -type f -name '*nix' | xargs sed -i '' 's/CC=cc/CC=\$\{stdenv\.cc\.targetPrefix\}cc/g'
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/luajit/default.nix [hunk 1]
```diff
@@ -66,7 +66,7 @@ stdenv.mkDerivation rec {
   ];
   makeFlags = [
     "PREFIX=$(out)"
-    "DEFAULT_CC=cc"
+    "DEFAULT_CC=${stdenv.cc.targetPrefix}cc"
```
Comment (siraben) [inline pkgs/development/interpreters/luajit/default.nix:69]: Need to revert

**Comment by siraben** [comment]
Due to the small number of rebuilds, I think it is OK to re-do this change to master.

**Comment by Mic92** [comment]
maybe also useful for nixpkgs-hammering? https://github.com/jtojnar/nixpkgs-hammering

**Comment by SuperSandro2000** [comment]
> maybe also useful for nixpkgs-hammering? [jtojnar/nixpkgs-hammering](https://github.com/jtojnar/nixpkgs-hammering)

@jtojnar @rmcgibbo

**Comment by rmcgibbo** [comment]
@siraben: Can you file an issue on hammering so this doesn't get lost?

**Comment by SuperSandro2000** [comment]
> @siraben: Can you file an issue on hammering so this doesn't get lost?

I am doing that.

**Comment by SuperSandro2000** [comment]
What about ``"AR=${stdenv.cc.targetPrefix}ar"``?

File: pkgs/data/misc/tzdata/default.nix [hunk 2]
```diff
@@ -45,7 +45,7 @@ stdenv.mkDerivation rec {
   preInstall = ''
      mv zic.o zic.o.orig
      mv zic zic.orig
-     make $makeFlags cc=cc AR=ar zic
+     make $makeFlags cc=${stdenv.cc.targetPrefix}cc AR=ar zic
```
Comment (SuperSandro2000) [inline pkgs/data/misc/tzdata/default.nix:48]:
```suggestion
     make $makeFlags cc=${stdenv.cc.targetPrefix}cc AR=${stdenv.cc.targetPrefix}ar zic
```

File: pkgs/development/libraries/java/junixsocket/default.nix [hunk 3]
```diff
@@ -23,7 +23,7 @@ stdenv.mkDerivation rec {
   ANT_ARGS =
     # Note that our OpenJDK on Darwin is currently 32-bit, so we have to build a 32-bit dylib.
     (if stdenv.is64bit then [ "-Dskip32=true" ] else [ "-Dskip64=true" ])
-    ++ [ "-Dgcc=cc" "-Dant.build.javac.source=1.6" ]
+    ++ [ "-Dgcc=${stdenv.cc.targetPrefix}cc" "-Dant.build.javac.source=1.6" ]
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/java/junixsocket/default.nix:26]: Not sure about this one.

File: pkgs/development/libraries/libfsm/default.nix [hunk 4]
```diff
@@ -19,7 +19,7 @@ stdenv.mkDerivation rec {
 
   # note: build checks value of '$CC' to add some extra cflags, but we don't
   # necessarily know which 'stdenv' someone chose, so we leave it alone (e.g.
-  # if we use stdenv vs clangStdenv, we don't know which, and CC=cc in all
+  # if we use stdenv vs clangStdenv, we don't know which, and CC=${stdenv.cc.targetPrefix}cc in all
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libfsm/default.nix:22]: This is a comment 😄

File: pkgs/tools/archivers/unar/default.nix [hunk 5]
```diff
@@ -15,7 +15,7 @@ stdenv.mkDerivation rec {
   postPatch = ''
     for f in Makefile.linux ../UniversalDetector/Makefile.linux ; do
       substituteInPlace $f \
-        --replace "= gcc" "=cc" \
+        --replace "= gcc" "=${stdenv.cc.targetPrefix}cc" \
         --replace "= g++" "=c++" \
```
Comment (SuperSandro2000) [inline pkgs/tools/archivers/unar/default.nix:19]:
```suggestion
        --replace "= g++" "=c++" \
```
```suggestion
        --replace "= g++" "=${stdenv.cc.targetPrefix}c++" \
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
> What about `AR=${stdenv.cc.targetPrefix}ar`, `${stdenv.cc.targetPrefix}strip`, `${stdenv.cc.targetPrefix}ranlib` ?

Addressed.

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/113295

6 packages built:
apg boxes hactool haproxy micronucleus mosml
```

**Review by Ericson2314 — APPROVED** [review]

File: pkgs/development/libraries/java/junixsocket/default.nix (hunk 3 above)
Comment (Ericson2314) [reply pkgs/development/libraries/java/junixsocket/default.nix:26]: I  think it's fine. :)

File: pkgs/games/crawl/default.nix [hunk 6]
```diff
@@ -44,7 +44,7 @@ stdenv.mkDerivation rec {
 
   fontsPath = lib.optionalString tileMode dejavu_fonts;
 
-  makeFlags = [ "prefix=${placeholder "out"}" "FORCE_CC=cc" "FORCE_CXX=c++" "HOSTCXX=c++"
+  makeFlags = [ "prefix=${placeholder "out"}" "FORCE_CC=${stdenv.cc.targetPrefix}cc" "FORCE_CXX=${stdenv.cc.targetPrefix}c++" "HOSTCXX=${stdenv.cc.targetPrefix}c++"
```
Comment (Ericson2314) [inline pkgs/games/crawl/default.nix:47]:
```suggestion
  makeFlags = [ "prefix=${placeholder "out"}" "FORCE_CC=${stdenv.cc.targetPrefix}cc" "FORCE_CXX=${stdenv.cc.targetPrefix}c++" "HOSTCXX=${buildPackages.stdenv.cc.targetPrefix}c++"
```
pretty sure `HOSTCXX` is the build->build one.

**Review by Ericson2314 — CHANGES_REQUESTED** [review]
Just the `HOSTCXX` one

File: pkgs/games/crawl/default.nix [hunk 7]
```diff
@@ -44,7 +44,7 @@ stdenv.mkDerivation rec {
 
   fontsPath = lib.optionalString tileMode dejavu_fonts;
 
-  makeFlags = [ "prefix=${placeholder "out"}" "FORCE_CC=cc" "FORCE_CXX=c++" "HOSTCXX=c++"
+  makeFlags = [ "prefix=${placeholder "out"}" "FORCE_CC=${stdenv.cc.targetPrefix}cc" "FORCE_CXX=${stdenv.cc.targetPrefix}c++" "HOSTCXX=${buildPackages.stdenv.cc.targetPrefix}c++"
```
Comment (Ericson2314) [inline pkgs/games/crawl/default.nix:47]: You need to import `buildPackages` too, right? :)

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>apg</li>
<li>boxes</li>
<li>hactool</li>
<li>haproxy</li>
<li>micronucleus</li>
<li>mosml</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
apg:
<details>


`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/tools/security/apg/default.nix:8:3:

```
  |
8 |   configurePhase = ''
  |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/security/apg/default.nix:14:5:

```
   |
14 |     ./apg.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/security/apg/default.nix:15:5:

```
   |
15 |     ./phony-install-target.patch
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
boxes:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/text/boxes/default.nix:43:5:

```
   |
43 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/text/boxes/default.nix:30:3:

```
   |
30 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
hactool:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/compression/hactool/default.nix:20:3:

```
   |
20 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
haproxy:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/networking/haproxy/default.nix:67:5:

```
   |
67 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
micronucleus:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/micronucleus/default.nix:32:5:

```
   |
32 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/micronucleus/default.nix:22:3:

```
   |
22 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
mosml:
<details>


`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/compilers/mosml/default.nix:25:5:

```
   |
25 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
perl is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/compilers/mosml/default.nix:7:3:

```
  |
7 |   buildInputs = [ gmp perl ];
  |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
</details>

**Pull request merged**